src/Facility.Core/ServiceObjectMessagePackFormatter.cs

```diff
@@ -0,0 +1,27 @@
+using System.Buffers;
+using MessagePack;
+using MessagePack.Formatters;
+
+namespace Facility.Core;
+
+public class ServiceObjectMessagePackFormatter : IMessagePackFormatter<ServiceObject>
+{
+	public void Serialize(ref MessagePackWriter writer, ServiceObject? value, MessagePackSerializerOptions options)
+	{
+		if (value is null)
+		{
+			writer.WriteNil();
+		}
+		else
+		{
+			var json = SystemTextJsonServiceSerializer.Instance.ToBytes(value.AsJsonObject());
```

Comment:
This is a hack. There should be a `IMessagePackFormatter<JsonNode>` implementation that knows how to serialize using string keys.